helix-core/src/syntax/tree_cursor.rs

```diff
@@ -204,7 +204,7 @@ impl<'a> TreeCursor<'a> {
 
         self.injection_ranges[start_idx..]
             .iter()
-            .take_while(|range| range.start < end)
+            .take_while(|range| range.end >= end)
```

Comment:
```suggestion
            .take_while(|range| range.start < end || range.depth > 1)
```

this should work

Reply:
It should be possible to set up some unit tests. We have a few at the bottom of `syntax` that set up a parser and then use it for a case